projects/portal/src/app/pages/delegate/validators/validators.usecase.service.ts

```diff
@@ -0,0 +1,166 @@
+import { CosmosRestService } from '../../../models/cosmos-rest.service';
+import { StoredWallet } from '../../../models/wallets/wallet.model';
+import { WalletService } from '../../../models/wallets/wallet.service';
+import {
+  validatorType,
+  validatorWithShareType,
+} from '../../../views/delegate/validators/validators.component';
+import { Injectable } from '@angular/core';
+import cosmosclient from '@cosmos-client/core';
+import {
+  InlineResponse20038DelegationResponses,
+  InlineResponse20041Validators,
+  InlineResponse20047,
+} from '@cosmos-client/core/esm/openapi';
+import { BehaviorSubject, combineLatest, Observable } from 'rxjs';
+import { concatMap, filter, map, mergeMap, withLatestFrom } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ValidatorsUseCaseService {
+  private validatorsList$: Observable<InlineResponse20041Validators[] | undefined>;
+  private allValidatorsTokens$: Observable<number | undefined>;
+  private validatorsWithShare$: Observable<validatorWithShareType[]>;
+  private accAddress$: Observable<cosmosclient.AccAddress>;
+  private delegationsInService$: Observable<InlineResponse20038DelegationResponses[] | undefined>;
+  private delegatedValidatorsInService$: Observable<
+    (InlineResponse20041Validators | undefined)[] | undefined
+  >;
+  private totalDelegation$: Observable<number | undefined>;
+  private unbondingDelegations$: Observable<(InlineResponse20047 | undefined)[]>;
```

Comment:
As a guideline, try to make local variables other than properties that are commonly used between methods.

Reply:
Thank you for review. deleted all local variables.